src/router/PrivateRoute.tsx

```diff
@@ -0,0 +1,20 @@
+import { useEffect } from "react";
+import { Outlet, useNavigate } from "react-router-dom";
+import { ROUTES } from "../constants/routes";
+import { useAppSelector } from "../store/store";
+import Login from "../pages/Login";
+
+const PrivateRoute = () => {
+  const navigate = useNavigate();
+  const { isLoggedIn } = useAppSelector((state) => state.auth);
```

Comment:
there's a cleaner way to pull state from redux. In your slice file, you could do something like this: `export selectIsLoggedIn = state=>state.isLoggedIn` and then you would use in this file like this: `const isLoggedIn=useAppSelector(selectIsLoggedIn)`